fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MVColumnPruner.java

```diff
@@ -93,6 +95,20 @@ public OptExpression visitLogicalTableScan(OptExpression optExpression, Void con
                 builder.withOperator(scanOperator);
                 LogicalScanOperator.Builder scanBuilder = (LogicalScanOperator.Builder) builder;
                 scanBuilder.setColRefToColumnMetaMap(newColumnRefMap);
+                ColumnRefSet outputRefSet = new ColumnRefSet(outputColumns);
+                outputRefSet.except(requiredOutputColumns);
+                if (!outputRefSet.isEmpty() && projection == null) {
+                    // should add a projection
+                    Map<ColumnRefOperator, ScalarOperator> projectionMap = Maps.newHashMap();
+                    for (ColumnRefOperator columnRefOperator : outputColumns) {
+                        if (outputRefSet.contains(columnRefOperator)) {
+                            continue;
+                        }
+                        projectionMap.put(columnRefOperator, columnRefOperator);
+                    }
+                    Projection newProjection = new Projection(projectionMap);
+                    builder.setProjection(newProjection);
+                }
```

Comment:
What about other operators?

Reply:
I add more uts to cover other operators. I found no problems